kylin/src/main/java/org/apache/zeppelin/kylin/KylinInterpreter.java

```diff
@@ -0,0 +1,197 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.zeppelin.kylin;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterPropertyBuilder;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Kylin interpreter for Zeppelin.
+ */
+public class KylinInterpreter extends Interpreter {
+  Logger logger = LoggerFactory.getLogger(KylinInterpreter.class);
+
+
+  static final String KYLIN_QUERY_API_URL = "kylin.api.url";
+  static final String KYLIN_USERNAME = "kylin.api.user";
+  static final String KYLIN_PASSWORD = "kylin.api.password";
+  static final String KYLIN_QUERY_PROJECT = "kylin.query.project";
+  static final String KYLIN_QUERY_OFFSET = "kylin.query.offset";
+  static final String KYLIN_QUERY_LIMIT = "kylin.query.limit";
+  static final String KYLIN_QUERY_ACCEPT_PARTIAL = "kylin.query.ispartial";
+  static final Pattern KYLIN_TABLE_FORMAT_REGEX_LABEL = Pattern.compile("\"label\":\"(.*?)\"");
+  static final Pattern KYLIN_TABLE_FORMAT_REGEX = Pattern.compile("\"results\":\\[\\[\"(.*?)\"]]");
+
+  static {
+    Interpreter.register(
+        "kylin",
+        "kylin",
+        KylinInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add(KYLIN_USERNAME, "ADMIN", "username for kylin user")
+            .add(KYLIN_PASSWORD, "KYLIN", "password for kylin user")
+            .add(KYLIN_QUERY_API_URL, "http://<host>:<port>/kylin/api/query", "Kylin API.")
+            .add(KYLIN_QUERY_PROJECT, "default", "kylin paroject name")
+            .add(KYLIN_QUERY_OFFSET, "0", "kylin query offset")
+            .add(KYLIN_QUERY_LIMIT, "5000", "kylin query limit")
+            .add(KYLIN_QUERY_ACCEPT_PARTIAL, "true", "The kylin query partial flag").build());
+  }
+
+
+
+
+  public KylinInterpreter(Properties property) {
+    super(property);
+  }
+  @Override
+  public void open() {
+
+  }
+
+  @Override
+  public void close() {
+
+  }
+
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context) {
+    try {
+      return executeQuery(st);
+    } catch (IOException e) {
+      logger.info("failed to query data in kylin ", e.getMessage());
```

Comment:
Maybe this should be an `error` or at least `warn`. And logging the stacktrace would be more useful for debugging issues, i.e. `logger.error("Failed to query data in kylin ", e);`